### ISSUE TITLE
Modal Base Style is adding a margin to the top of a "full" size modal but shouldn't

### ISSUE THREAD

**Issue by timkindberg** [chakra-ui/chakra-ui#3962]
This line: https://github.com/chakra-ui/chakra-ui/blob/14b5a422d87d77ea3842f5480e3270273bb42c04/packages/theme/src/components/modal.ts#L39

Is causing the full size modal to be "shifted down". "full" size modals need to remove the margin that the base style adds.

![image](https://user-images.githubusercontent.com/168046/117204445-4cdee680-adbe-11eb-9873-8779e661c401.png)

**Comment by chr-ge** [comment]
That margin is there on purpose. It's a design choice. If you want you can remove it.

**Comment by timkindberg** [comment]
I think you may be wrong though. Because the modal hangs off the bottom of the viewport and the buttons are just out of view, you have to scroll just a bit to reveal them. If this is intentional I'll be surprised.

**Comment by chr-ge** [comment]
My guess is the margin is there so that it's clear that a modal is open and that it can be closed by clicking the overlay (along with the close button). 
Some people might be confused if their entire screen is covered with a white box?

**Comment by timkindberg** [comment]
Sorry to rebuttal twice in a row here... but if you just go play with it you'll see it's super odd. https://chakra-ui.com/docs/overlay/modal#modal-sizes 

There is no way this is intentional. 

However I will say that the concept of leaving some room to see the overlay is a good one though and personally I've modified this modal in my own project to look like this, there is a consistent 1rem margin around the whole thing:
![image](https://user-images.githubusercontent.com/168046/118022511-6fc34a80-b32a-11eb-9b77-ce0f84cf3e42.png)

I checked with @with-heart and he thinks it might be wrong as well.
